includes/classes/RegisteredDataHandler.php

```diff
@@ -0,0 +1,637 @@
+						}
+					}
+					$current_extra_data = $extra_data[ $index ] ?? array();
+					$replacement        = call_user_func_array( $callback_fn, array( $current_extra_data, $source_data, $post_data, $this->connection_data ) );
+					if ( ! empty( $replacement ) ) {
+						foreach ( $block_attribute as $attribute ) {
+							if ( isset( $replacement[ $attribute ] ) ) {
+								$block['attrs'][ $attribute ] = $replacement[ $attribute ];
+							}
+						}
+
+						// Do replacement for innerHTML if it's set.
+						if ( ! empty( $replacement['inner_content_replacements'] ) ) {
+							$block = $this->search_replace_block_inner_content( $block, $replacement['inner_content_replacements'] );
+						}
+
+						$modified = true;
+					}
+					$index++;
+				} elseif ( isset( $block['attrs'][ $block_attribute ] ) ) {
+					$source_data        = $block['attrs'][ $block_attribute ];
+					$current_extra_data = $extra_data[ $index ] ?? array();
+					$replacement        = call_user_func_array( $callback_fn, array( $current_extra_data, $source_data, $post_data, $this->connection_data ) );
+					if ( ! empty( $replacement ) ) {
+						$block['attrs'][ $block_attribute ] = $replacement;
+
+						// Handle inner content replacements for media blocks.
+						$type = $registered_data['type'] ?? '';
+						if ( 'media' === $type && ! empty( $replacement ) ) {
+							$from_url = $current_extra_data['url'] ?? '';
+							$to_url   = wp_get_attachment_url( $replacement );
+
+							if ( ! empty( $from_url ) && ! empty( $to_url ) ) {
+								$replacements = array(
+									array(
+										'search'  => $from_url,
+										'replace' => $to_url,
+									),
+									array(
+										'search'  => 'wp-image-' . $source_data,
+										'replace' => 'wp-image-' . $replacement,
+									),
+								);
+								// Try replacing the guid as well, due to the media url could be different based on from where it's being pulled.
+								if ( ! empty( $current_extra_data['guid'] ) ) {
+									$replacements[] = array(
+										'search'  => $current_extra_data['guid'],
+										'replace' => $to_url,
+									);
+								}
+								$block = $this->search_replace_block_inner_content( $block, $replacements );
+							}
+						}
+
+						$modified = true;
+					}
+					$index++;
+				}
+			}
+
+			if ( ! empty( $block['innerBlocks'] ) && is_array( $block['innerBlocks'] ) ) {
+				$inner_result = $this->process_blocks_data_recursive( $block['innerBlocks'], $registered_data, $extra_data, $post_data, $index );
+				if ( $inner_result['modified'] ) {
+					$block['innerBlocks'] = $inner_result['blocks'];
+					$modified             = true;
+				}
+			}
+		}
+
+		return array(
+			'blocks'   => $blocks,
+			'modified' => $modified,
+		);
+	}
+
+	/**
+	 * This function processes the registered data for the post content and post meta.
+	 * It calls the callback function provided in the registered data and updates the post data.
```

Comment:
Nitpick: No need to document that the function is a function 😉 (Applies in a few places)

```suggestion
	 * Processes the registered data for the post content and post meta.
	 *
	 * Calls the callback function provided in the registered data and updates the post data.
```

Reply:
I apologize on behalf of the cursor 😄.  Handled in https://github.com/10up/distributor/pull/1309/commits/fb1f75905de91fbb103eb54ca5ead88d910a3aae